src/common/socket.h

```diff
@@ -62,6 +65,25 @@ inline std::string GetHostName() {
   return std::string(buf.c_str());
 }
 
+/*!
+ * \brief ValidateIP validates an ip address.
+ * \param ip The ip address in string format localhost or x.x.x.x format
+ * \return result of operation.
+ */
+inline bool ValidateIP(std::string ip) {
+  if (ip == "localhost") {
+    return true;
+  }
+  std::vector<std::string> list = Split(ip, '.');
+  if (list.size() != 4)
+    return false;
+  for (std::string str : list) {
```

Comment:
when iterate over a list, use const std::string& str(otherwise could result in a copy)